lambdas/process-candidate-identity/src/main/java/uk/gov/di/ipv/core/processcandidateidentity/ProcessCandidateIdentityHandler.java

```
@@ -375,28 +382,21 @@ private JourneyResponse getJourneyResponseForProfileMatching(
             IpvSessionItem ipvSessionItem,
             ClientOAuthSessionItem clientOAuthSessionItem,
             String deviceInformation,
-            String ipAddress,
             List<VerifiableCredential> sessionVcs,
             AuditEventUser auditEventUser)
-            throws HttpResponseExceptionWithErrorBody, CiRetrievalException, ParseException {
+            throws HttpResponseExceptionWithErrorBody, CiRetrievalException, ParseException,
+                    NoCriForIssuerException, CredentialParseException {
 
         var areVcsCorrelated = userIdentityService.areVcsCorrelated(sessionVcs);
 
         if (!areVcsCorrelated) {
             return JOURNEY_VCS_NOT_CORRELATED;
         }
 
-        // This is a performance optimisation to save a call to CIMIT if it is not required
-        // If the VTR only contains one entry then it is impossible for a user to reach here
-        // with a breaching CI so we don't have to check.
         var contraIndicators =
-                clientOAuthSessionItem.getVtr().size() == 1
-                        ? List.<ContraIndicator>of()
-                        : cimitService.getContraIndicators(
-                                clientOAuthSessionItem.getUserId(),
-                                clientOAuthSessionItem.getGovukSigninJourneyId(),
-                                ipAddress,
-                                ipvSessionItem);
+                cimitService.getContraIndicatorsFromVc(
+                        ipvSessionItem.getSecurityCheckCredential(),
+                        clientOAuthSessionItem.getUserId());
```

theoretically, we should always have a cimit vc by this point (bc we call it as part of check-existing-identity + after every cri call) but should we still check if there isn't one stored in-session?

In this case, how should it be handled? Should we throw an error?

---

It would probably be nice to handle it with a BAD_REQUEST response (like we do for missing session id)

---

thanks :) have added to process-candidate-identity and build-user-identity lambdas